### PR TITLE
[WIP] Broader config file support

### PR DESCRIPTION
**Pull request by pingsutw**
Signed-off-by: Kevin Su <pingsutw@apache.org>

# TL;DR
`Setup-config` command in flytekit to use the YAML format so that you can run setup-config with flytekit and then run flytectl.
`flyte-cli` load config in YAML format.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Add an option that lets the user choose the output format.
To support backward compatibility, we could output both `ini` and `yaml` format. 

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1035

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit/clis/flyte_cli/main.py [hunk 1]
```diff
@@ -485,6 +487,7 @@ def _render_schedule_expr(lp):
 _output_location_prefix_option = _click.option(
     "-o", "--output-location-prefix", help="Custom output location prefix for offloaded types (files/schemas)"
 )
+_output_config_format = _click.option("-o", "--format", type=str, default="ini", help="Output config format (ini/yaml)")
```
Comment (katrogan) [inline flytekit/clis/flyte_cli/main.py:490]: can we use click.Choice here? https://click.palletsprojects.com/en/8.0.x/options/#choice-options

Comment (pingsutw) [reply flytekit/clis/flyte_cli/main.py:490]: sure, thnaks.

**Comment by wild-endeavor** [comment]
hey sorry should've included more information in the issue.  this won't work unfortunately.  the reason is that not only is flytectl's format different, the keys are entirely different - the names i mean.  for instance
```
admin:
  endpoint: dns:///fakeflyte.com
  insecure: false
  authType: Pkce
logger:
  show-source: true
  level: 1
```

**Comment by wild-endeavor** [comment]
let's think about this in the broader context of updating our configuration support.  The UX definitely needs improvement in a number of places.

**Comment by pingsutw** [comment]
> hey sorry should've included more information in the issue. this won't work unfortunately. the reason is that not only is flytectl's format different, the keys are entirely different - the names i mean. for instance

Okay, I see

**Comment by wild-endeavor** [comment]
closing in favor of https://github.com/flyteorg/flytekit/pull/898 sorry
